### ISSUE TITLE
Proposal: new no-fallthrough option - "allowEmptyCase"

### ISSUE THREAD

**Issue by zkldi** [eslint/eslint#14146]
**What rule do you want to change?**

no-fallthrough

**Does this change cause the rule to produce more or fewer warnings?**

Fewer warnings, as it allows a case that is disallowed currently.

**How will the change be implemented? (New option, new default behavior, etc.)?**

New option, this option should not be enabled by default or in the recommended, as it could cause regressions.

**Please provide some example code that this change will affect:**

```js
let foo = 2;

// lets declare a switch case that does stuff if foo is any of a set of values
switch (foo) {
    case 1: // this currently causes an error due to no-fallthrough working as intended
    case 49: // with allowEmptyCase, this will not cause an error.
    case 101:
    case 2:
        dostuff();
        break;
    default:
        break;
}

// if you have commentPattern on, this can work, but it is dense:
switch (foo) {
    case 1:
        // break omitted
    case 49:
        // break omitted
    case 101:
        // break omitted
    case 2:
        dostuff();
        break;
    default:
        break;
}
```

**What does the rule currently do for this code?**

No-fallthrough currently disallows cases that do not have a break statement. While it allows a comment pattern for declaring explicit break omission, it gets a bit tedious for repeated cases.

**What will the rule do after it's changed?**

If enabled, the option will cause no-fallthrough to allow fallthrough **if and only if** the case has no content.

The justification for this rule is that ``no-fallthrough`` is legitimately useful and catches accidental fallthrough, so opting
out of no-fallthrough is not ideal. However, the legitimate use of case fallthrough in switch statements has to be repeatedly opted into with comments, which results in incredibly dense code.

Given that empty cases have no legitimate use in switch statements (you can't really accidentally have fallthrough)
and empty cases are quite commonly used in sequence to allow fallthrough of multiple results, I think this rule makes sense.

```js
switch (foo) {
    case 1: // allowed with allowEmptyCase: true
    case 2:
        dostuff();
        break;
    default:
        break;
}

switch (foo) {
    case 1: 
        // allowed with allowEmptyCase: true, as comments are not content
        //  - also allowed with commentPattern, though, so this is kind of redundant.
    case 2:
        dostuff();
        break;
    default:
        break;
}

switch (foo) {
    case 1: 
        // NOT allowed with allowEmptyCase: true, as this case contains content, 
        // and therefore the missing break could be a legitimate mistake.
        doOtherStuff();
    case 2:
        dostuff();
        break;
    default:
        break;
}
```

``allowEmptyCase: false`` is the default, and has no effect versus the default behaviour.

**Are you willing to submit a pull request to implement this change?**

Yes, I am willing to work on this change, but I have not read the eslint codebase yet, so I do not know how difficult it will be to implement.

**Comment by zkldi** [comment]
Update: after checking the code and using the demo, it appears this is the default behaviour, and another plugin I had installed was causing this to error erroneously.
